packages/react-shared-contexts/src/TooltipContext/TooltipContext.ts

```diff
@@ -0,0 +1,54 @@
+import * as React from 'react';
+import { useFluent } from '../ProviderContext/ProviderContext';
+import { TooltipManager, TooltipManagerCreateFunction } from './TooltipManager.types';
+
+/**
+ * The context provided by TooltipProvider
+ *
+ * {@docCategory TooltipProvider}
+ */
+export type TooltipContext = {
+  /**
+   * The instance of TooltipManager
+   */
+  tooltipManager?: TooltipManager;
+
+  /**
+   * The number of components using the TooltipManager.
+   * This is internal to useTooltipManager and shouldn't be otherwise modified.
+   */
+  useCount: number;
+};
+
+/**
+ * Context shared by all of the tooltips in the app
+ */
+export const TooltipContext = React.createContext<TooltipContext>({ useCount: 0 });
+
+/**
+ * Get the TooltipManager from the TooltipContext, and create it if this is the first component that's using it.
+ *
+ * @param createTooltipManager - Function to create the TooltipManager if it doesn't exist yet
+ */
+export const useTooltipManager = (createTooltipManager: TooltipManagerCreateFunction) => {
+  const { targetDocument } = useFluent();
+
+  const context = React.useContext(TooltipContext);
+  if (!context.tooltipManager) {
+    context.tooltipManager = createTooltipManager(targetDocument);
+  }
+
+  React.useEffect(() => {
+    context.useCount++;
+
+    return () => {
+      context.useCount--;
+      if (context.useCount === 0) {
+        context.tooltipManager?.destroy();
+        context.tooltipManager = undefined;
+      }
+    };
+  }, [context]);
```

Comment:
I would be careful there as it's safe until we will not have an another item in `TooltipContext` than can cause a recreation of a context value. To clarify I mean this case:

```tsx
const tooltipUseCount = useConst<TooltipContext>(() => ({ useCount: 0 }));
const tooltipAnotherDep = ""; /* something that changes time to time */

const tooltipContext = React.useMemo(
  () => ({ useCount, dep: tooltipAnotherDep }),
  [
    tooltipUseCount,
    tooltipAnotherDep
  ] /* 👆 as this can change, a context value will change and will trigger destroying `tooltipManager`  */
);
```

---

_I don't have an actionable suggestion there, just sharing an observation._ May be we can add some comments there and into `useFluentProvider.ts` to provide some context?

Reply:
I'll think about a better way to handle this and address it in a future PR (or possibly this PR).